src/query/procedure/mg_procedure_impl.cpp

```diff
@@ -1846,10 +1846,9 @@ mgp_error mgp_vertex_label_at(mgp_vertex *v, size_t i, mgp_label *result) {
                       "Expected LabelToName to return a pointer or reference, so we "
                       "don't have to take a copy and manage memory.");
 
-        const auto &name = std::visit([label](const auto *impl) { return impl->LabelToName(label); }, v->graph->impl);
-        return name.c_str();
+        return std::visit([label](const auto *impl) { return impl->LabelToName(label).data(); }, v->graph->impl);
```

Comment:
`data()` doesn't guarantee that the string is null terminated, thus `c_str()` has to be used.